apps/api/src/controllers/posts.ts

```diff
@@ -1,155 +1,91 @@
+import Post from '../models/post';
+
 type Post = { title: string; description: string; category: string | object; image?: string; comments?: string[] };
 
 const POST_NO_FOUND = 'Post not found';
 const INVALID_POST = 'Invalid post missing requirement (title / description / category)';
 const INVALID_COMMENT = 'Invalid comment missing content';
-// variable for manage the info
-const posts = [];
-
-const getCategory = (category: string) => {
-  return posts.filter((post) => post.category === category);
-};
-const getPostId = (id: string) => {
-  return posts.find((post) => post.id === id);
-};
-
-const getIndexById = (array: any[], id: string) => {
-  return array?.length > 0 ? array.findIndex((item) => item.id === id) : -1;
-};
-
-const isPostValid = (postObj: { title: string; description: string; category: string }) => {
-  const { title, description, category } = postObj;
-
-  const isTitleValid = title?.trim()?.length > 0;
-  const isDescValid = description?.trim()?.length > 0;
-  const isCategoryValid = category?.trim()?.length > 0;
-
-  return isTitleValid && isDescValid && isCategoryValid;
-};
-
-const createNewPost = (post: Post) => {
-  const { title, description, category, image, comments } = post;
-
-  const postCategory = {
-    id: category,
-    name: 'some category'
-  };
-  const newPost = {
-    id: Date.now().toString(),
-    title,
-    description,
-    category: postCategory,
-    image: image ?? '',
-    comments: comments?.length ?? []
-  };
-
-  return newPost;
-};
-
-const createNewComment = (comment: { content: string; author?: string }) => {
-  const { author, content } = comment;
-  const newComment = {
-    id: Date.now().toString(),
-    author: author ?? 'anonymous',
-    content
-  };
-  return newComment;
-};
-
-const updatePostAttributes = (updates: Post, id: string) => {
-  const { title, image, description, category } = updates;
-  const postIndex = getIndexById(posts, id);
-  if (postIndex < 0) {
-    return { isInvalid: true, post: {} };
-  }
-  let post = { ...posts[postIndex] };
-  if (title) {
-    post.title = title;
-  }
-  if (image) {
-    post.image = image;
-  }
-  if (description) {
-    post.description = description;
-  }
-  if (category) {
-    post.category.name = category;
-  }
-
-  posts[postIndex] = post;
-
-  return { isInvalid: false, post };
-};
 
 // - `GET /posts` Return an array of all the posts with status code 200
-const getAllPosts = (req, res) => {
-  res.status(200).json(posts);
+const getAllPosts = async (req, res) => {
+  try {
+    const categories = await Post.find({});
```

Comment:
```suggestion
    const categories = await Post.find().populate('category');
```

We need to populate the category here, same as you did in the posts by category so the front end can display the name of the category instead of the id.